R/github.R

```diff
@@ -40,114 +30,162 @@ get_github_user <- function(token) {
 #' @description This is a function to get the GitHub user's info
 #' @param token You can provide the Personal Access Token key directly or this function will attempt to grab a PAT that was stored using the `authorize("github")` function
 #' @return Information regarding a github account
-#' @importFrom utils menu installed.packages
-#' @importFrom httr oauth_app oauth_endpoints oauth2.0_token
 #' @export
 #' @examples \dontrun{
 #'
 #' authorize("github")
 #' get_github_user()
 #' }
 get_github_user <- function(token) {
-
   if (is.null(token)) {
     # Get auth token
     token <- get_token(app_name = "github")
-  } else {
-    token <- token
   }
 
-    get_github(
+  get_github(
     url = "https://api.github.com/user",
-    token = token,
+    token = token
   )
 }
 
-#' Get the repository info
-#' @description This is a function to get the GitHub user's info
+#' Retrieve list of repositories for an owner
+#' @description This is a function to get the information about a repository
 #' @param token You can provide the Personal Access Token key directly or this function will attempt to grab a PAT that was stored using the `authorize("github")` function
-#' @param owner The owner
-#' @param repo The repository
+#' @param owner The owner of the repository. So for `https://github.com/fhdsl/metricminer`, it would be `fhdsl`
+#' @param repo The repository name. So for `https://github.com/fhdsl/metricminer`, it would be `metricminer`
+#' @param count The number of responses that should be returned. Default is 20 or you can say "all" to retrieve all.
 #' @return Information regarding a github account
-#' @importFrom utils menu installed.packages
-#' @importFrom httr oauth_app oauth_endpoints oauth2.0_token
+#' @importFrom gh gh
 #' @export
 #' @examples \dontrun{
 #'
 #' authorize("github")
-#' get_github_repo()
+#' get_repo_list(owner = "fhdsl")
 #' }
-get_github_repo <- function(token,  owner, repo) {
+#'
+get_repo_list <- function(owner = NULL, count = "all", token = NULL) {
```

Comment:
```suggestion
get_repo_list <- function(owner, count = "all", token = NULL) {
```

Got rid of default `NULL` for the argument `owner` to force the user to specify that argument (or else the function won't run)